test/test_sox_compatibility.py

```diff
@@ -9,9 +9,27 @@
 
 
 @common_utils.skipIfNoSoxBackend
-class TestFunctionalFiltering(common_utils.TorchaudioTestCase):
+class TestFunctionalFiltering(common_utils.TempDirMixin, common_utils.TorchaudioTestCase):
     backend = 'sox'
 
+    def setUp(self):
+        common_utils.TempDirMixin.setUp(self)
+        common_utils.TorchaudioTestCase.setUp(self)
+
+        NOISE_SAMPLE_RATE = 44100
+        INT16_MAX = 32767
+        noise_waveform = common_utils.get_whitenoise(
+            sample_rate=NOISE_SAMPLE_RATE, duration=5
+        )
+        # convert to int16, scipy wav writer automatically selects wav format based on tensor dtype
+        noise_waveform_as_int = torch.as_tensor(
+            INT16_MAX * noise_waveform, dtype=torch.int16
+        )
+        self.noise_filepath = self.get_temp_path("whitenoise.wav")
+        common_utils.save_wav(
+            self.noise_filepath, noise_waveform_as_int, NOISE_SAMPLE_RATE
+        )
```

Comment:
As far as I understand, the current implementation runs test this way

```
1. Generate a white noise
 |
 v
2. Save the noise to file --------> 3. Run sox effect from the file
 |                                            |
 | 4.1. Load data from the file               |
 v                                            v
4.2. Run functional on loaded data -------> compare    
```

This process has a duplicated data preparation at `1.` and `4.1.`. The result of data loaded from the file at `4.1.` is already available at `1`, so it's redundant. And `4.1.` requires a loading function, which is `torchaudio.load` currently. This has a problem because the test now depends on both `torchaudio.load` and functionals being tested. When test is implemented this way and the test fails it is not immediately clear which caused a problem, and the worst case, both implementation are wrong but somehow the test passes so that we do not notice the actual issue. To avoid that we should avoid using `torchaudio.load` in this test suite. We could use `common_utils.load_wav` but simplifying the process to generate synthetic data allows us to do it without it.

```
1. Generate a white noise ---------------> 2.1. Save the noise to file
|                                            |
v                                            v
3. Run functional the generated data.         2.2. Run sox effect from the file
|                                            |
-----------------> compare <------------------
```

Reply:
Sure, the only caveat is you'll have to generate two versions of the same signal, a `float32` and `int16` version, or we do the conversion outside the `get_whitenoise` function.  Thoughts?